InformationScripting/src/queries/CompositeQuery.cpp

```diff
@@ -76,7 +77,13 @@ QList<TupleSet> CompositeQuery::execute(QList<TupleSet> input)
 					return m.outputFrom_ == currentNode && m.outputIndex_ == outIndex;
 				});
 				Q_ASSERT(inputIt != receiver->inputMap_.end());
-				auto output = hasOutput ? currentNode->calculatedOutputs_[outIndex] : TupleSet();
+				TupleSet output;
+				if (hasOutput)
+				{
+					// early abort in case of error:
+					if (currentNode->calculatedOutputs_[outIndex]) output = currentNode->calculatedOutputs_[outIndex].value();
+					else return { currentNode->calculatedOutputs_[outIndex].error() };
```

Comment:
If we allow parallel execution we will have to rewrite this code anyway. Therefore I will simply return the error Optional if there is an error.

I'm not sure why QList<Optional<.. would hinder me from doing this?